stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+    The Prior generator function.
+    The arguments of the prior function are in the order of
+        Kernel arguments (RN arguments, QPO arguments),
+        Mean arguments
+        Miscellaneous arguments
+
+    Examples
+    --------
+    A prior function for a Red Noise kernel and a Gaussian mean function
+    Obain the parameters list
+    if not can_sample:
+        pytest.skip("Jaxns not installed. Cannot make jaxns specific prior.")
+    if not tfp_available:
+        pytest.skip("Tensorflow probability required to make priors.")
+
+    params_list = get_gp_params("RN", "gaussian")
+
+    Make a prior dictionary using tensorflow_probability distributions
+    prior_dict = {
+       "log_A": tfpd.Uniform(low = jnp.log(1e-1), high = jnp.log(2e+2)),
+       "t0": tfpd.Uniform(low = 0.0 - 0.1, high = 1 + 0.1),
+       "log_sig": tfpd.Uniform(low = jnp.log(0.5 * 1 / 20), high = jnp.log(2) ),
+       "log_arn": tfpd.Uniform(low = jnp.log(0.1) , high = jnp.log(2) ),
+       "log_crn": tfpd.Uniform(low = jnp.log(1 /5), high = jnp.log(20)),
+    }
+
+    prior_model = get_prior(params_list, prior_dict)
+
+    """
+    if not jax_avail:
+        raise ImportError("Jax is required")
+
+    if not can_sample:
+        raise ImportError("Jaxns not installed. Cannot make jaxns specific prior.")
+
+    if not tfp_available:
+        raise ImportError("Tensorflow probability required to make priors.")
+
+    def prior_model():
+        prior_list = []
+        for i in params_list:
+            if isinstance(prior_dict[i], tfpd.Distribution):
+                parameter = yield Prior(prior_dict[i], name=i)
+            else:
+                parameter = yield prior_dict[i]
+            prior_list.append(parameter)
+        return tuple(prior_list)
+
+    return prior_model
+
+
+def get_likelihood(params_list, kernel_type, mean_type, **kwargs):
+    """
+    A log likelihood generator function based on given values.
+    Makes a jaxns specific log likelihood function which takes in the
+    parameters in the order of the parameters list, and calculates the
+    log likelihood of the data given the parameters, and the model
+    (kernel, mean) of the GP model. **Note** Any parameters with a prefix
+    of "log_" are taken to be in the log scale.
+
+    Parameters
+    ----------
+    params_list:
+        A list in order of the parameters to be used.
+
+    prior_dict:
+        A dictionary of the priors of parameters to be used.
+
+    kernel_type:
+        The type of kernel to be used in the model.
+
+    mean_type:
+        The type of mean to be used in the model.
+
+    **kwargs:
+        The keyword arguments to be used in the log likelihood function.
+        **Note**: The keyword arguments Times and counts are necessary for
+        calculating the log likelihood.
+        Times: np.array or jnp.array
+            The time array of the lightcurve
+        counts: np.array or jnp.array
+            The photon counts array of the lightcurve
```

Comment:
If these are required parameters, why are they hidden in `**kwargs`, should they not be be upfront as required arguments?

Reply:
Also, here and in the code, `Times` should be lowercase

Reply:
Yes, they were necessary parameters so putting them in **kwargs was not required. Have made them as arguments and changed the Docstring.